source/dem/disable_particle_contact.cc

```diff
@@ -0,0 +1,334 @@
+  solid_fractions.reinit(triangulation->n_active_cells());
+  active_ghost_cells.clear();
+
+  // Iterating through the active cells in the triangulation
+  for (const auto &cell : background_dh.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() || cell->is_ghost())
+        {
+          // Get active and ghost cells in a set
+          active_ghost_cells.insert(cell);
+          double granular_temperature_cell = 0.0;
+          double solid_fraction            = 0.0;
+
+          // Particles in the cell
+          typename Particles::ParticleHandler<dim>::particle_iterator_range
+                             particles_in_cell = particle_handler.particles_in_cell(cell);
+          const unsigned int n_particles_in_cell =
+            particle_handler.n_particles_in_cell(cell);
+
+          // Check if the cell has any particles
+          if (n_particles_in_cell > 0)
+            {
+              // Initialize variables for average velocity
+              Tensor<1, dim> velocity_cell_sum;
+              Tensor<1, dim> velocity_cell_average;
+
+              // Initialize variables for void fraction
+              double       solid_volume = 0.0;
+              const double cell_volume  = cell->measure();
+
+              // Initialize velocity fluctuations
+              Tensor<1, dim> cell_velocity_fluctuation_squared_sum;
+              Tensor<1, dim> cell_velocity_fluctuation_squared_average;
+
+              // First loop over particles in cell to calculation the average
+              // velocity and the void fraction
+              for (typename Particles::ParticleHandler<
+                     dim>::particle_iterator_range::iterator
+                     particles_in_cell_iterator = particles_in_cell.begin();
+                   particles_in_cell_iterator != particles_in_cell.end();
+                   ++particles_in_cell_iterator)
+                {
+                  auto &particle_properties =
+                    particles_in_cell_iterator->get_properties();
+
+                  for (int d = 0; d < dim; ++d)
+                    {
+                      velocity_cell_sum[d] +=
+                        particle_properties[DEM::PropertiesIndex::v_x + d];
+                    }
+
+                  solid_volume +=
+                    M_PI *
+                    pow(particle_properties[DEM::PropertiesIndex::dp], dim) /
+                    (2.0 * dim);
+                }
+
+              // Calculate average velocity in the cell
+              for (int d = 0; d < dim; ++d)
+                velocity_cell_average[d] =
+                  velocity_cell_sum[d] / n_particles_in_cell;
+
+              // Calculate void fraction of cell
+              solid_fraction = solid_volume / cell_volume;
+
+              // Second loop over particle to calculate the average granular
+              // temperature
+              for (typename Particles::ParticleHandler<
```

Comment:
Can't you do this in a single loop?

Reply:
No since I need the velocity cell average before